src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/MinorAlleleFractionCache.java

```diff
@@ -0,0 +1,64 @@
+package org.broadinstitute.hellbender.tools.exome.allelefraction;
+
+import org.apache.commons.math3.optim.MaxEval;
+import org.apache.commons.math3.optim.nonlinear.scalar.GoalType;
+import org.apache.commons.math3.optim.univariate.BrentOptimizer;
+import org.apache.commons.math3.optim.univariate.SearchInterval;
+import org.apache.commons.math3.optim.univariate.UnivariateObjectiveFunction;
+import org.broadinstitute.hellbender.tools.exome.AllelicCount;
+import org.broadinstitute.hellbender.utils.GATKProtectedMathUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A helper class to maintain a cache of computed maximum-likelihood estimates of minor allele fraction
+ * for {@link AllelicCount#estimateMinorAlleleFraction(double)}.
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class MinorAlleleFractionCache {
+    // constants for Brent optimization of minor allele fraction
+    private static final MaxEval BRENT_MAX_EVAL = new MaxEval(100);
+    private static final double MINOR_ALLELE_FRACTION_RELATIVE_TOLERANCE = 0.00001;
+    private static final double MINOR_ALLELE_FRACTION_ABSOLUTE_TOLERANCE = 0.00001;
+    private static final BrentOptimizer OPTIMIZER =
+            new BrentOptimizer(MINOR_ALLELE_FRACTION_RELATIVE_TOLERANCE, MINOR_ALLELE_FRACTION_ABSOLUTE_TOLERANCE);
+
+    private static final Map<MinorAlleleFractionCacheKey, Double> cache = new HashMap<>();
+
+    public static double get(final int a, final int r, final double allelicBias) {
+        final MinorAlleleFractionCacheKey key = new MinorAlleleFractionCacheKey(a, r, allelicBias);
+        if (cache.containsKey(key)) {
+            return cache.get(key);
+        } else {
+            final double minorAlleleFraction = calculateMinorAlleleFraction(a, r, allelicBias);
+            cache.put(key, minorAlleleFraction);
+            return minorAlleleFraction;
+        }
+    }
+
+    //See docs/CNVs/CNV-methods.pdf for derivation of likelihood
+    private static double calculateMinorAlleleFraction(final int a, final int r, final double bias) {
```

Comment:
At some expense in run-time (mitigated by the caching), a slightly more right way of doing this is to get the whole `AlleleFractionState` from the `AlleleFractionInitializer` and then use `AlleleFractionModeller::collapsedHetLogLikelihood` as the maximization objective.  This corresponds to marginalizing the allelic bias while accounting for outlier probability instead of pegging allelic bias to its mean.  While this should improve accuracy a bit, the main benefit in my opinion is the use of a single likelihood method.  In my opinion it's worth it despite the run time.